TetrisEngine.java

```diff
@@ -0,0 +1,134 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+
+public class TetrisEngine {
+	private static final String INPUT_FILE = "input.txt";
+	private static final String OUTPUT_FILE = "output.txt";
+	//private static final int GRID_SIZE = 100;
+	
+	private static HashMap<Character, TetrisPiece> PieceMap = new HashMap<Character, TetrisPiece>();
+	private static TetrisGrid Grid = new TetrisGrid();
+	
+	public static void main(String[] args) {
+		PopulatePieceMap();
+		List<String> inputLines = ReadFile();
+		String output = "";
+
+		for(String line : inputLines)
+		{
+			line = line.trim();
+			output += DoTetris(line);
+		}
+		WriteFile(output);
+	}
+	
+	private static String DoTetris(String input)
+	{
+		String out = "";
+		Grid.Reset();
+		for(String shapeStr : input.split(","))
+		{			
+			if(shapeStr.length() != 2)
+				return "Invalid Input";
+			
+			Grid.AddPiece(MakePiece(shapeStr.charAt(0), Character.getNumericValue(shapeStr.charAt(1))));
+		}
+		Grid.toString();
+		out = String.format("%s\n", Grid.getHeight());
+		return out;
+	}
+	
+	private static TetrisPiece MakePiece(char shape, int index)
+	{		
+		TetrisPiece newPiece = new TetrisPiece(PieceMap.get(shape), index);				
+		return newPiece;
+	}
+	
+	private static List<String> ReadFile()
+	{
+		List<String> input = null;
+		try {
+			
+			input = Files.readAllLines(Paths.get(INPUT_FILE));
+
+		} catch (IOException e) {
+
+		}
+		
+		return input;
+	}
+	
+	private static void WriteFile(String out)
+	{
+
+		try (Writer writer = new BufferedWriter(new OutputStreamWriter(
+	              new FileOutputStream(OUTPUT_FILE), "utf-8"))) {
+			writer.write(out);
+		} catch (UnsupportedEncodingException e) {

```

Comment:
Since these are all handled the same, you can catch them all in a single block: https://www.journaldev.com/629/java-catch-multiple-exceptions-rethrow-exception